Sources/Queues/QueuesConfiguration.swift

```diff
@@ -24,50 +45,61 @@ public struct QueuesConfiguration {
     }
 
     /// Sets the number of workers used for handling jobs.
-    public var workerCount: WorkerCount
+    public var workerCount: WorkerCount {
+        get { self.dataBox.withLockedValue { $0.workerCount } }
+        set { self.dataBox.withLockedValue { $0.workerCount = newValue } }
+    }
     
     /// A logger
     public let logger: Logger
     
     // Arbitrary user info to be stored
-    public var userInfo: [AnyHashable: Any]
+    public var userInfo: [AnySendableHashable: any Sendable] {
+        get { self.dataBox.withLockedValue { $0.userInfo } }
+        set { self.dataBox.withLockedValue { $0.userInfo = newValue } }
+    }
     
-    var jobs: [String: AnyJob]
-    var scheduledJobs: [AnyScheduledJob]
-    var notificationHooks: [JobEventDelegate]
+    var jobs: [String: any AnyJob] {
+        get { self.dataBox.withLockedValue { $0.jobs } }
+        set { self.dataBox.withLockedValue { $0.jobs = newValue } }
+    }
     
-    /// Creates an empty `JobsConfig`
+    var scheduledJobs: [AnyScheduledJob] {
+        get { self.dataBox.withLockedValue { $0.scheduledJobs } }
+        set { self.dataBox.withLockedValue { $0.scheduledJobs = newValue } }
+    }
+    
+    var notificationHooks: [any JobEventDelegate] {
+        get { self.dataBox.withLockedValue { $0.notificationHooks } }
+        set { self.dataBox.withLockedValue { $0.notificationHooks = newValue } }
+    }
+    
+    /// Creates an empty ``QueuesConfiguration``.
     public init(
         refreshInterval: TimeAmount = .seconds(1),
         persistenceKey: String = "vapor_queues",
         workerCount: WorkerCount = .default,
         logger: Logger = .init(label: "codes.vapor.queues")
     ) {
-        self.jobs = [:]
-        self.scheduledJobs = []
         self.logger = logger
         self.refreshInterval = refreshInterval
         self.persistenceKey = persistenceKey
         self.workerCount = workerCount
-        self.userInfo = [:]
-        self.notificationHooks = []
     }
     
-    /// Adds a new `Job` to the queue configuration.
-    /// This must be called on all `Job` objects before they can be run in a queue.
+    /// Adds a new ``Job`` to the queue configuration.
     ///
-    /// - Parameter job: The `Job` to add.
-    mutating public func add<J>(_ job: J)
-        where J: Job
-    {
-        self.logger.trace("Adding job type: \(J.name)")
+    /// This must be called on all ``Job`` objects before they can be run in a queue.
+    ///
+    /// - Parameter job: The ``Job`` to add.
+    mutating public func add<J: Job>(_ job: J) {
```

Comment:
```suggestion
    mutating public func add(_ job: some Job) {
```

Reply:
Can't, we have to refer to `J` in the method body.